namer/namer.cc

```diff
@@ -1353,6 +1353,7 @@ class SymbolDefiner {
             }
         }
         sym = ctx.state.enterStaticFieldSymbol(ctx.locAt(staticField.lhsLoc), scope, name);
```

Comment:
Can we grep for calls to `enterStaticFieldSymbol` and see if it suggests other code paths that we need to write tests for?

For example, I think that

```ruby
@@x = 1
```

will also suffer from this crash but exercise a different code path.

Another one I think suffers the same fate is

```
X = type_template
```

Reply:
This test with a comment manipulation passes on master
```ruby
class B
extend T::Sig
  @@x = 1

  sig {void}
  def self.foo
    T.reveal_type(@@x) # error: Revealed type: `Integer`
  end
end
```

Reply:
And this test fails even with a change from this PR
```ruby
class B
  extend T::Generic
  X = type_template
end

T.reveal_type(B::X) # error: Revealed type: `Runtime object representing type: T.class_of(B)::X`
```

It would require a separate fix

Reply:
The test case with `@@x` updates the location correctly, because it was already handled:
https://github.com/sorbet/sorbet/blob/24163761b87fc5ac0a40b31b567384c2a8688d1a/namer/namer.cc#L1317-L1319